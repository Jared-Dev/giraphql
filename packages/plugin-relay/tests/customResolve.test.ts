import { execute } from 'graphql';
import { gql } from 'graphql-tag';
import schema from './examples/custom-resolve/schema';

describe('custom node resolve', () => {
  it('resolves a node with a global id', async () => {
    const query = gql`
      query {
        node(id: "VXNlcjo4") {
          ... on User {
            age
          }
        }
      }
    `;

    const result = await execute({
      schema,
      document: query,
      contextValue: {},
    });

    expect(result).toMatchSnapshot();
  });
});
