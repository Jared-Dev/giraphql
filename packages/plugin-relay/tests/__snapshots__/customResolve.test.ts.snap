// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`custom node resolve > resolves a node with a global id 1`] = `
{
  "data": {
    "node": {
      "age": 20,
    },
  },
}
`;
